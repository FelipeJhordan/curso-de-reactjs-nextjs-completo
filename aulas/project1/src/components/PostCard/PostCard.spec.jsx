const { render, screen } = require('@testing-library/react');
const { PostCard } = require('.');
const { postCardPropsMock } = require('./mock');

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard content={props} />);
    expect(screen.getByRole('img', { name: 'title 1' })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: /title 1/i })).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard content={props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
