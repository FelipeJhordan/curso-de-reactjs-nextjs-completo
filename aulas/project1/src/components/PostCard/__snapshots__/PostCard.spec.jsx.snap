// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <div
    class="post__content"
  >
    <img
      alt="title 1"
      src="img/img.png"
    />
    <h2>
      title 1
    </h2>
    <p>
      body 1
    </p>
  </div>
</div>
`;
